app/assets/javascripts/discourse/tests/acceptance/topic-user-status-test.js

```diff
@@ -0,0 +1,126 @@
+import {
+  acceptance,
+  exists,
+  publishToMessageBus,
+  query,
+  queryAll,
+} from "discourse/tests/helpers/qunit-helpers";
+import { visit } from "@ember/test-helpers";
+import { test } from "qunit";
+import TopicFixtures from "discourse/tests/fixtures/topic";
+
+acceptance("Topic - User Status", function (needs) {
+  const status = { emoji: "tooth", description: "off to dentist" };
+
+  needs.user();
+  needs.pretender((server, helper) => {
+    server.get("/t/299/1.json", () => {
+      const response = TopicFixtures["/t/299/1.json"];
+      response.post_stream.posts.forEach((post) => {
+        post.user_status = status;
+      });
+
+      return helper.response(200, response);
+    });
+  });
+
+  test("shows user status next to avatar on posts", async function (assert) {
+    this.siteSettings.enable_user_status = true;
+    await visit("/t/-/299/1");
+
+    assert.equal(
+      queryAll(".topic-post .user-status-message").length,
+      3,
+      "all posts has user status"
+    );
+  });
+});
+
+acceptance("Topic - User Status - live updates", function (needs) {
+  const userId = 1;
+  const status = { emoji: "tooth", description: "off to dentist" };
+
+  needs.user();
+  needs.pretender((server, helper) => {
+    server.get("/t/299/1.json", () => {
+      const response = TopicFixtures["/t/299/1.json"];
```

Comment:
aren't we supposed to cloneJson this to avoid any side effect?

Reply:
We are! Thank you for catching this.